pkg/deploy/service.go

```diff
@@ -290,6 +295,40 @@ func buildStrategy(modelStrategy model.OrchestrationConfig, strategyName string)
 	return nil, fmt.Errorf("%s is not a valid strategy; define canary or blueGreen strategy", strategyName)
 }
 
+func createTrafficManagement(mo *model.OrchestrationConfig, currentTarget string) (*[]de.KubernetesV2TrafficManagement, error) {
+	if mo.TrafficManagement == nil {
+		return nil, nil
+	}
+	var tms []de.KubernetesV2TrafficManagement
+	for _, tm := range *mo.TrafficManagement {
+		for _, t := range tm.Targets {
+			if t == currentTarget && tm.SMI != nil {
+				var configErrors []string
+				if tm.SMI.RootServiceName == nil {
+					configErrors = append(configErrors,"rootServiceName required in smi")
+				}
+				if tm.SMI.CanaryServiceName == nil {
```

Comment:
Both `canaryServiceName` and `trafficSplitName` are optional: https://armory.slab.com/posts/tdd-11-smi-traffic-management-bt031ao0#he5vp-api-contract